src/notify/subscription/handlers.go

```diff
@@ -0,0 +1,81 @@
+package subscription
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"notify/jsonUtils"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+// Create is a REST API for creating a new subscription, based on the JSON payload
+func Create(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+	subscription := Subscription{}
+
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&subscription)
+	if err != nil {
+		jsonUtils.Error(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	err = subscription.insert()
+	if err != nil {
+		jsonUtils.Error(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	subscriptionJSON, _ := json.Marshal(subscription)
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(201)
+	fmt.Fprintf(w, "%v", string(subscriptionJSON))
```

Comment:
These lines can be replaced with ```jsonUtils.Output(w, 201, subscription)```.  See the event handlers.


(Same with the others below.)